activation/activation.go

```diff
@@ -735,20 +756,61 @@ func (b *Builder) createAtx(
 		}
 	}
 
-	atx := wire.ActivationTxV1{
-		InnerActivationTxV1: wire.InnerActivationTxV1{
-			NIPostChallengeV1: *challenge,
-			Coinbase:          b.Coinbase(),
-			NumUnits:          nipostState.NumUnits,
-			NIPost:            wire.NiPostToWireV1(nipostState.NIPost),
-		},
-	}
-	if nonce != nil {
-		atx.VRFNonce = (*uint64)(nonce)
+	coinbase := b.Coinbase()
+
+	switch version {
+	case types.AtxV1:
+		atx := wire.ActivationTxV1{
+			InnerActivationTxV1: wire.InnerActivationTxV1{
+				NIPostChallengeV1: *wire.NIPostChallengeToWireV1(challenge),
+				Coinbase:          coinbase,
+				NumUnits:          nipostState.NumUnits,
+				NIPost:            wire.NiPostToWireV1(nipostState.NIPost),
+			},
+		}
+		if nonce != nil {
+			atx.VRFNonce = (*uint64)(nonce)
+		}
+		atx.Sign(sig)
+
+		return &atx, nil
+	case types.AtxV2:
+		atx := &wire.ActivationTxV2{
+			PublishEpoch:   challenge.PublishEpoch,
+			PositioningATX: challenge.PositioningATX,
+			Coinbase:       &coinbase,
+			NiPosts: []wire.NiPostsV2{
+				{
+					Membership: wire.MerkleProofV2{
+						Nodes:       nipostState.Membership.Nodes,
+						LeafIndices: []uint64{nipostState.Membership.LeafIndex},
+					},
+					Challenge: types.Hash32(nipostState.NIPost.PostMetadata.Challenge),
+					Posts: []wire.SubPostV2{
+						{
+							Post:     *wire.PostToWireV1(nipostState.Post),
```

Comment:
It's a bit confusing that a `V1` function is used for `V2` 😅

Reply:
Then the naming is a bit confusing? How about `wire.PostToWire`?

Reply:
But it is converting between types.Post and wire.PostV1.